enhancements/storage/storage-security-policy.md

```diff
@@ -0,0 +1,198 @@
+
+Having said that, `fsGroupChangePolicy` still defaults to `Always`, which means kubelet will always recursively change permissions. The reason for defaulting to `Always` is - mostly backward compatibility and if users bring in volumes that were used outside kubernetes. In these cases `OnRootMismatch` may not be a safe default.
+
+But for most intent and purpose, it can be a reasonable default and hence this proposal allows admins to configure a namespace such that, all pods created in the namespace will default to `OnRootMismatch` policy if pod does not specify a policy of its own. If a pod specifies its own `fsGroupChangePolicy` that is still used as a default and will not be overridden.
+
+### Motivation for selinuxChangePolicy defaulting
+
+https://github.com/kubernetes/enhancements/issues/1710 is bringing changes which can be breaking for certain pods which use different selinux contexts for same volume between different pods or same pod.
+
+This will speed up time spent while recursively changing selinux policies of a volume.
+
+By allowing cluster-admins to set namespace default of `selinuxChangePolicy` we want to give control over how these changes are applied to Openshift clusters, without explicitly  changing pods themselves.
+
+### User Stories
+
+N/A
+
+### Goals
+
+- Allow Openshift admins to configure per-namespace default of `fsGroupChangePolicy`.
+- Allow Openshift admins to configure per-namespace default of `selinuxChangePolicy`.
+
+### Non-Goals
+
+* Change Openshift default globally.
+
+## Proposal
+
+### Allow admins to opt-in to `fsGroupChangePolicy` via namespace label
+
+We propose a label `storage.openshift.io/fsgroup-change-policy` if that is set to `OnRootMismatch` in a namespace, then Openshift's SCC hooks start defaulting all pods created in that namespace with `OnRootMismatch` `FSGroupChangePolicy`.
+
+```go
+func getPodFsGroupChangePolicy(ns *corev1.Namespace) *api.PodFSGroupChangePolicy {
+    fsGroupPolicy, ok := ns.Labels[securityv1.FSGroupChangePolicy]
+    if !ok {
+        return nil
+    }
+    if fsGroupPolicy == "OnRootMismatch" {
+        onRootMismatchPolicy := api.FSGroupChangeOnRootMismatch
+        return &onRootMismatchPolicy
+    }
+    return nil
+}
+```
+
+### Allow admins to opt-in to `selinuxChangePolicy` via namespace label
+
+Similar to mechanism proposed above,  we propose to use label `storage.openshift.io/selinux-change-policy` to define namespace wide `selinuxChangePolicy` if none are specified in a pod.
+
+### Setting of `fsGroupChangePolicy` and `selinuxChangePolicy` during admission
+
+We propose to add a new admission hook (which will be eventually replaced with MAP) which will trigger the defautling logic, similar to how SCC matching hooks are triggered.
```

Comment:
What's the timeline for this feature?

Do we have an existing admission hook where this code could/should live? Or is this net new?